lib/utils/tunnel.ts

```diff
@@ -96,11 +97,14 @@ const openPortThroughProxy = (
 	}
 
 	return new Promise<Socket>((resolve, reject) => {
-		const proxyTunnel = new Socket();
-		proxyTunnel.on('error', reject);
-		proxyTunnel.connect(proxyPort, proxyServer, () => {
-			const proxyConnectionHandler = (data: Buffer) => {
-				proxyTunnel.removeListener('data', proxyConnectionHandler);
+		const proxyTunnel = tls.connect(proxyPort, proxyServer, {}, () => {
+			if (!proxyTunnel.authorized) {
+				reject(proxyTunnel.authorizationError);
+				return;
+			}
+
+			const proxyDataHandler = (data: Buffer) => {
+				proxyTunnel.removeListener('data', proxyDataHandler);
```

Comment:
Does this `removeListener` call do the same as using `proxyTunnel.once(...)` instead of `proxyTunnel.on(...)`?

Reply:
@pdcastro actually yes, it should, probably a cleaner way to do it also 👍🏻